docs/src/logging.md

```diff
@@ -0,0 +1,145 @@
+
+with_logger(logger) do
+    @warn "Memory usage high"  # Written to rank-specific log file
+end
+```
+
+### MPILogger
+
+`MPILogger(context)` adds an MPI rank prefix to all log messages:
+
+```julia
+using ClimaComms, Logging
+ctx = ClimaComms.context()
+logger = MPILogger(ctx)
+
+with_logger(logger) do
+    @info "Processing data..."  # Output: [P1] Processing data...
+end
+```
+
+## Log Levels
+
+Julia provides four standard log levels, in order of increasing severity:
+
+1. `Debug`: Detailed information for debugging
+2. `Info`: General information about program execution
+3. `Warn`: Warnings about potential issues
+4. `Error`: Error conditions that might still allow the program to continue
+
+See [Julia documentation](https://docs.julialang.org/en/v1/stdlib/Logging/#Log-event-structure) for more detailed information.
+
+You can define custom log levels using `LogLevel`:
+
+```julia
+const Trace = LogLevel(-1000)  # Lower number = less severe
+@macroexpand @logmsg Trace "Very detailed trace message"
+```
+
+To disable all log messages at log levels equal to or less than a given `LogLevel`, use [`Logging.disable_logging(level)`](https://docs.julialang.org/en/v1/stdlib/Logging/#Logging.disable_logging).
+
+## Filtering Log Messages
+
+[LoggingExtras.jl](https://github.com/JuliaLogging/LoggingExtras.jl) provides powerful filtering capabilities through the `EarlyFilteredLogger(filter, logger)`, which takes two arguments:
+
+- `filter(log_args)` is a function which takes in `log_args` and returns a Boolean determining if the message should be logged. `log_args` is a NamedTuple with fields `level`, `_module`, `id` and `group`. Example `filter` functions are provided below in the Common Use Cases.
+- `logger` is any existing logger, such as `Logging.ConsoleLogger()` or `MPILogger(ctx)`.
+
+### Common Use Cases
+
+#### How do I save log output to stdout and a file simultaneously?
+
+`LoggingExtras.TeeLogger(loggers)` composes multiple loggers, allowing for multiple loggers at once as shown below. Note that `ClimaComms.MPIFileLogger` provides this functionality as well.
+
+```julia
+using Logging, LoggingExtras
+
+io = open("simulation.log", "w")
+loggers = (Logging.ConsoleLogger(stdout), Logging.ConsoleLogger(io))
+tee_logger = LoggingExtras.TeeLogger(loggers)
+
+with_logger(tee_logger) do
+    @warn "Log to stdout and file"
+end
+
+close(io)
+```
+
+#### How do I filter out warning messages?
+
+```julia
+using Logging, LoggingExtras
+
+function no_warnings(log_args)
+    return log_args.level != Logging.Warn
+end
+
+filtered_logger = EarlyFilteredLogger(no_warnings, Logging.current_logger())
+
+with_logger(filtered_logger) do
+    @warn "Hide this warning"
+    @info "Display this message"
+end
+```
+
+#### How do I filter out messages from certain modules?
+
+```julia
+using Logging, LoggingExtras
+
+function module_filter(excluded_modules)
+    return function(log_args)
+        !(log_args._module in excluded_modules)
+    end
```

Comment:
I never realized you could write anonymous functions like this.

Maybe we can add a line to explain this block a little further?

Reply:
I'm a bit confused by this myself. This function returns a function? I'm wondering what the utility is, also, I'd probably just write it as:

```julia
module_filter(excluded_modules) = log_args ->
    !(log_args._module in excluded_modules)
```

Reply:
Also, should we make this an `@example`?

Reply:
Or is that not a good idea for some reason?